lib/ansible/modules/cloud/azure/azure_rm_keyvault.py

```diff
@@ -0,0 +1,430 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Ian Philpot, <ian.philpot@microsoft.com>;
+# Obezimnaka Boms, <t-ozboms@microsoft.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_keyvault
+
+version_added: "2.5"
+
+short_description: Create, delete and update key vaults.
+
+description:
+    - Creates, deletes, and updates key vaults.
+
+options:
+    resource_group:
+        description:
+            - name of resource group.
+        required: true
+    name:
+        description:
+            - name of the key vault.
+        required: true
+    state:
+        description:
+            - Assert the state of the key vault. Use 'present' to create or update and
+              'absent' to delete.
+        default: present
+        choices:
+            - absent
+            - present
+    enabled_for_deployment:
```

Comment:
I'm thinking instead of the 3 `enabled_for_*` options there should be 1 option like `resource_access` (can be any name) that takes in a list of values like `vm`, `disk_encryption`, `resource_manager` which then dictate whether they are set. This is just a suggestion but I think it should be discussed at the Azure WG to get a group consensus.

As a side note, whenever you have a boolean option you should set it like

```
bool_options:
  description:
  - Description for this option.
  type: bool
  default: 'no'  # can be 'yes'
```

The Ansible doc generator will automatically add the yes/no options to the choices and format it accordingly.